lib/grape/middleware/base.rb

```diff
@@ -21,9 +21,17 @@ def call(env)
 
       def call!(env)
         @env = env
+        error_class = nil
         before
-        @app_response = @app.call(@env)
-        after || @app_response
+        begin
+          @app_response = @app.call(@env)
+        rescue => e
+          error_class = e
+        ensure
+          after_response = after
+        end
+
+        after_response || (error_class ? (fail error_class) : @app_response)
       end
```

Comment:
Makes sense.
So should I remove Response from [this line](https://github.com/ruby-grape/grape/blob/master/lib/grape/middleware/base.rb#L36)?